ufersa/plataforma-sabia
# Feature/Web/Protected Components

## Summary

A component to protect another components from unauthorized access.

Resolves #119.

## Relevant technical choices

- By default, the component protects children only from not logged in users:
```jsx
const App = () => {
	<Protected>
		<h1>Protected Children</h1>
	</Protected>
};
```

- We can also protect children based on authorization using the `role` prop and pass a URL to redirect to another page:
```jsx
const App = () => {
	<Protected role="admin" redirectTo="/about">
		<h1>Protected Children</h1>
	</Protected>
};
```

## QA Steps

Only check if user is logged in:

![basic usage](https://user-images.githubusercontent.com/30202634/84557069-232c5500-acfe-11ea-8dff-e88e3a5007b3.gif)

Pass a needed user role:

![user role](https://user-images.githubusercontent.com/30202634/84557142-fcbae980-acfe-11ea-8141-448817b2379f.gif)

Redirect to another page:

![redirect](https://user-images.githubusercontent.com/30202634/84557129-cf6e3b80-acfe-11ea-9114-134c3a81c40e.gif)

## Checklist
